teamcity/resource_build_config.go

```diff
@@ -826,11 +826,19 @@ func expandBuildFeature(raw interface{}) (api.BuildFeature, error) {
 func expandBuildSteps(list interface{}) ([]api.Step, error) {
 	out := make([]api.Step, 0)
 	in := list.([]interface{})
+	names := make(map[string]bool)
 	for _, i := range in {
 		s, err := expandBuildStep(i)
 		if err != nil {
 			return nil, err
 		}
+		_, exist := names[s.GetName()]
```

Comment:
Since `exist` is only needed for the if check, it'd be better to keep it in the if-block's scope like so:

```golang
if _, exist := names[s.GetName()]; exist {
    ...
} else {
    names[s.GetName()] = struct{}{}
}
```

(alternatively, if you decide to keep it as a `map[string]bool`, you can just use the fact that map misses return the zero-value, and the zero-value of booleans is `false`:

```golang
if names[s.GetName()] {
    ...
} else {
    ...
}
```

Reply:
new commit uploaded